pkg/gohai/platform/platform_linux.go

```diff
@@ -5,16 +5,81 @@
 
 package platform
 
-import "strings"
-
-var unameOptions = []string{"-s", "-n", "-r", "-m", "-p", "-i", "-o"}
-
-func updateArchInfo(archInfo map[string]string, values []string) {
-	archInfo["kernel_name"] = values[0]
-	archInfo["hostname"] = values[1]
-	archInfo["kernel_release"] = values[2]
-	archInfo["machine"] = values[3]
-	archInfo["processor"] = values[4]
-	archInfo["hardware_platform"] = values[5]
-	archInfo["os"] = strings.Trim(values[6], "\n")
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
+	"golang.org/x/sys/unix"
+)
+
+// getUnameOS is similar to `uname -o`
+//
+// uname uses preprocessor macros to determine the printable name of the operating system,
+// cf https://github.com/coreutils/gnulib/blob/master/m4/host-os.m4
+//
+// on every Linux device it prints "GNU/Linux", probably because uname is from GNU coreutils,
+// so if you are using it your OS is considered to be GNU
+func getUnameOS() string {
+	// eventually we might want to return different values depending on the actual OS
+	// (not all Linux are GNU)
+	return "GNU/Linux"
+}
+
+// check if the vendor is AMD
+// reader should be an io.Reader over /proc/cpuinfo
+func isVendorAMD(reader io.Reader) bool {
```

Comment:
Golang usually starts with the mehods name and are supposed to be a short sentence.

```suggestion
// isVendorAMD checks if the vendor is AMD.
// The reader is expecetd be an io.Reader over /proc/cpuinfo
func isVendorAMD(reader io.Reader) bool {
```